### PR TITLE
Labels update

### LINKED ISSUE #1244: Add labels to grid renderables
Labels should be added to the new grid renderables `RenderableGrid` and `RenderableRadialGrid`. The first one can then replace the grids showing the different distances, which are now created using RenderableDUMeshes. 

OBS! This issue should also include structuring the different units for scales in a good way, or even some generalization of the labels. (this is currently done using essentially copy pasted code for all label implementations...). Maybe implement a `LabelComponent` class? The files `distanceconversion.h`and `distanceconstants.h` can be useful for this.

### PR DESCRIPTION
**Pull request by Roxeena**
Adds labels to the grids (fixes #1244) and centralizes the logic regarding the labels into one `LabelsComponent` in the space module. 

The `RenderableLabels` class has been renamed to `RenderableLabel` since it only creates **one** label.

There will be some more work in the future on the labels to fix some of their rendering issues, but that will be a separate PR.

### CODE REVIEW

File: modules/base/rendering/grids/renderableboxgrid.h [hunk 1]
```diff
@@ -71,6 +73,11 @@ class RenderableBoxGrid : public Renderable {
 
     GLenum _mode = GL_LINE_STRIP;
     std::vector<Vertex> _varray;
+
+    // Labels
+    bool _hasLabels = false;
+    properties::BoolProperty _drawLabels;
+    std::unique_ptr<LabelsComponent> _labels = nullptr;
```
Comment (alexanderbock) [inline modules/base/rendering/grids/renderableboxgrid.h:80]: No need to default initialize it;  the default constructor does it to `nullptr`

File: modules/base/rendering/grids/renderablegrid.cpp [hunk 2]
```diff
@@ -81,6 +81,19 @@ namespace {
         "This value species the size of each dimensions of the grid"
     };
 
+    constexpr openspace::properties::Property::PropertyInfo DrawLabelInfo = {
+        "DrawLabels",
+        "Draw Labels",
+        "Determines whether labels should be drawn or hidden"
+    };
+
+    static const openspace::properties::PropertyOwner::PropertyOwnerInfo LabelsInfo =
```
Comment (alexanderbock) [inline modules/base/rendering/grids/renderablegrid.cpp:90]: `constexpr`

File: modules/base/rendering/grids/renderablegrid.cpp [hunk 3]
```diff
@@ -155,10 +176,31 @@ RenderableGrid::RenderableGrid(const ghoul::Dictionary& dictionary)
...
+
+        _labels = std::make_unique<LabelsComponent>(*p.labels);
+        _hasLabels = true;
+        addPropertySubOwner(_labels.get());
+    }
 }
 
 bool RenderableGrid::isReady() const {
-    return _gridProgram != nullptr;
+    bool isReady = _gridProgram != nullptr;
```
Comment (alexanderbock) [inline modules/base/rendering/grids/renderablegrid.cpp:191]: Same as above

File: modules/base/rendering/grids/renderableradialgrid.cpp [hunk 4]
```diff
@@ -71,6 +71,19 @@ namespace {
         "ring"
     };
 
+    constexpr openspace::properties::Property::PropertyInfo DrawLabelInfo = {
+        "DrawLabels",
+        "Draw Labels",
+        "Determines whether labels should be drawn or hidden"
+    };
+
+    static const openspace::properties::PropertyOwner::PropertyOwnerInfo LabelsInfo =
```
Comment (alexanderbock) [inline modules/base/rendering/grids/renderableradialgrid.cpp:80]: -> `constexpr`

File: modules/base/rendering/grids/renderableradialgrid.cpp [hunk 5]
```diff
@@ -134,10 +155,31 @@ RenderableRadialGrid::RenderableRadialGrid(const ghoul::Dictionary& dictionary)
...
+        _hasLabels = true;
+        addPropertySubOwner(_labels.get());
+    }
 }
 
 bool RenderableRadialGrid::isReady() const {
-    return _gridProgram != nullptr;
+    bool isReady = _gridProgram != nullptr;
+
+    if (_hasLabels) {
```
Comment (alexanderbock) [inline modules/base/rendering/grids/renderableradialgrid.cpp:172]: Same as above

File: modules/base/rendering/grids/renderableradialgrid.h [hunk 6]
```diff
@@ -85,6 +87,11 @@ class RenderableRadialGrid : public Renderable {
 
     std::vector<GeometryData> _circles;
     GeometryData _lines{GL_LINES};
+
+    // Labels
+    bool _hasLabels = false;
+    properties::BoolProperty _drawLabels;
+    std::unique_ptr<LabelsComponent> _labels = nullptr;
```
Comment (alexanderbock) [inline modules/base/rendering/grids/renderableradialgrid.h:94]: No need to allocate `nullptr` to it;  the default constructor does the right thing

File: modules/base/rendering/grids/renderablesphericalgrid.cpp [hunk 7]
```diff
@@ -55,6 +55,19 @@ namespace {
         "This value specifies the line width of the spherical grid"
     };
 
+    constexpr openspace::properties::Property::PropertyInfo DrawLabelInfo = {
+        "DrawLabels",
+        "Draw Labels",
+        "Determines whether labels should be drawn or hidden"
+    };
+
+    static const openspace::properties::PropertyOwner::PropertyOwnerInfo LabelsInfo =
```
Comment (alexanderbock) [inline modules/base/rendering/grids/renderablesphericalgrid.cpp:64]: -> `constexpr`

File: modules/digitaluniverse/rendering/renderablebillboardscloud.cpp [hunk 8]
```diff
@@ -140,6 +112,13 @@ namespace {
         "Determines whether labels should be drawn or hidden"
     };
 
+    static const openspace::properties::PropertyOwner::PropertyOwnerInfo LabelsInfo =
```
Comment (alexanderbock) [inline modules/digitaluniverse/rendering/renderablebillboardscloud.cpp:115]: -> `constexpr`

File: modules/digitaluniverse/rendering/renderablebillboardscloud.h [hunk 9]
```diff
@@ -133,20 +125,19 @@ class RenderableBillboardsCloud : public Renderable {
...
     std::string _datavarSizeOptionString;
 
     DistanceUnit _unit = DistanceUnit::Parsec;
 
     speck::Dataset _dataset;
-    speck::Labelset _labelset;
     speck::ColorMap _colorMap;
 
+    // Everything related to the labels are handled by LabelsComponent
+    std::unique_ptr<LabelsComponent> _labels = nullptr;
```
Comment (alexanderbock) [inline modules/digitaluniverse/rendering/renderablebillboardscloud.h:139]: No need for `nullptr` assignment here

File: modules/digitaluniverse/rendering/renderableplanescloud.cpp [hunk 10]
```diff
@@ -67,44 +65,11 @@ namespace {
...
-        "The minimal size (in pixels) of the text for the labels for the astronomical "
-        "objects being rendered"
-    };
-
-    constexpr openspace::properties::Property::PropertyInfo LabelMaxSizeInfo = {
-        "TextMaxSize",
-        "Text Max Size",
-        "The maximum size (in pixels) of the text for the labels for the astronomical "
-        "objects being rendered"
+    static const openspace::properties::PropertyOwner::PropertyOwnerInfo LabelsInfo =
```
Comment (alexanderbock) [inline modules/digitaluniverse/rendering/renderableplanescloud.cpp:68]: -> `constexpr`

File: modules/digitaluniverse/rendering/renderableplanescloud.h [hunk 11]
```diff
@@ -113,20 +102,20 @@ class RenderablePlanesCloud : public Renderable {
...
     std::filesystem::path _texturesPath;
     std::string _luminosityVar;
 
     DistanceUnit _unit = DistanceUnit::Parsec;
 
     speck::Dataset _dataset;
-    speck::Labelset _labelset;
+
+    // Everything related to the labels are handled by LabelsComponent
+    std::unique_ptr<LabelsComponent> _labels = nullptr;
```
Comment (alexanderbock) [inline modules/digitaluniverse/rendering/renderableplanescloud.h:118]: Same here

File: modules/space/labelscomponent.h [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+#include <openspace/properties/vector/ivec2property.h>
+#include <openspace/properties/vector/vec3property.h>
+#include <openspace/util/distanceconversion.h>
+#include <ghoul/glm.h>
+#include <filesystem>
+
+namespace ghoul::fontrendering { class Font; }
+
+namespace openspace {
+    struct RenderData;
```
Comment (alexanderbock) [inline modules/space/labelscomponent.h:42]: One level of indentation too much

File: modules/space/rendering/renderableconstellationsbase.cpp [hunk 13]
```diff
@@ -94,63 +57,35 @@ namespace {
...
-        "The unit used for the label data"
-    };
-
     constexpr openspace::properties::Property::PropertyInfo SelectionInfo = {
         "ConstellationSelection",
         "Constellation Selection",
         "The constellations that are selected are displayed on the celestial sphere"
     };
 
+    static const openspace::properties::PropertyOwner::PropertyOwnerInfo LabelsInfo =
```
Comment (alexanderbock) [inline modules/space/rendering/renderableconstellationsbase.cpp:66]: -> `constexpr`

File: modules/space/rendering/renderableconstellationsbase.h [hunk 14]
```diff
@@ -108,16 +105,8 @@ class RenderableConstellationsBase : public Renderable {
...
-    std::shared_ptr<ghoul::fontrendering::Font> _font = nullptr;
-    DistanceUnit _labelUnit = DistanceUnit::Parsec;
-    properties::Vec3Property _textColor;
-    properties::FloatProperty _textOpacity;
-    properties::FloatProperty _textSize;
-    properties::IVec2Property _textMinMaxSize;
-
-    properties::OptionProperty _renderOption;
+    // Everything related to the labels are handled by LabelsComponent
+    std::unique_ptr<LabelsComponent> _labels = nullptr;
```
Comment (alexanderbock) [inline modules/space/rendering/renderableconstellationsbase.h:109]: No need for `nullptr` assignment

**Review by alexanderbock — APPROVED** [review]
:+1:

File: modules/base/rendering/grids/renderableboxgrid.cpp [hunk 15]
```diff
@@ -54,6 +54,19 @@ namespace {
         "This value species the size of each dimensions of the box"
     };
 
+    constexpr openspace::properties::Property::PropertyInfo DrawLabelInfo = {
+        "DrawLabels",
+        "Draw Labels",
+        "Determines whether labels should be drawn or hidden"
+    };
+
+    static const openspace::properties::PropertyOwner::PropertyOwnerInfo LabelsInfo =
```
[the remark this answers is not in the document]
Comment (Roxeena) [reply modules/base/rendering/grids/renderableboxgrid.cpp:63]: Hmm, the compiler doesn't like `constexpr` for `PropertyOwnerInfo`s and I can't see anywhere else in the code where we use this combination 🤔

[the remark this answers is not in the document]
Comment (alexanderbock) [reply modules/base/rendering/grids/renderableboxgrid.cpp:63]: Yea, you're right;  looked at the class right now and it's indeed using dynamic memory... so never mind

**Pull request merged**